backends/sqlite/sqlite_backend_test.go

```diff
@@ -0,0 +1,837 @@
+			Queue: queue,
+			Payload: map[string]interface{}{
+				"message": fmt.Sprintf("hello world: %d", internal.RandInt(10000000000)),
+			},
+		})
+		if e != nil || jid == jobs.DuplicateJobID {
+			t.Error(e)
+		}
+
+		jid2, e := nq.Enqueue(ctx, &jobs.Job{
+			Queue: queue2,
+			Payload: map[string]interface{}{
+				"message": fmt.Sprintf("hello world: %d", internal.RandInt(10000000000)),
+			},
+		})
+		if e != nil || jid2 == jobs.DuplicateJobID {
+			t.Error(e)
+		}
+	}()
+
+results_loop:
+	for {
+		select {
+		case <-timeoutTimer:
+			err = jobs.ErrJobTimeout
+			break results_loop
+		case <-done:
+			doneCnt++
+			if doneCnt == 2 {
+				break results_loop
+			}
+		}
+	}
+
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCron(t *testing.T) {
+	done := make(chan bool, 1)
+	defer close(done)
+	const cron = "* * * * * *"
+	connString, _ := prepareAndCleanupDB(t)
+
+	ctx := context.TODO()
+	nq, err := neoq.New(ctx,
+		neoq.WithBackend(sqlite.Backend),
+		sqlite.WithConnectionString(connString),
+		neoq.WithLogLevel(logging.LogLevelDebug),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer nq.Shutdown(ctx)
+
+	h := handler.NewPeriodic(func(ctx context.Context) (err error) {
+		done <- true
+		return
+	})
+
+	h.WithOptions(
+		handler.JobTimeout(500*time.Millisecond),
+		handler.Concurrency(1),
+	)
+
+	err = nq.StartCron(ctx, cron, h)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// allow time for listener to start
+	time.Sleep(5 * time.Millisecond)
+
+	select {
+	case <-time.After(1 * time.Second):
+		err = errPeriodicTimeout
+	case <-done:
+	}
+
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestMultipleCronNodes(t *testing.T) {
+	jobsProcessed := sync.Map{}
+	const cron = "* * * * * *"
+	connString, _ := prepareAndCleanupDB(t)
+
+	workers := make([]neoq.Neoq, ConcurrentWorkers)
+	var jobsCompleted uint32
+	var duplicateJobs uint32
+	for i := 0; i < ConcurrentWorkers; i++ {
+		ctx := context.TODO()
+		nq, err := neoq.New(ctx,
+			neoq.WithBackend(sqlite.Backend),
+			sqlite.WithConnectionString(connString),
+			neoq.WithLogLevel(logging.LogLevelDebug),
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Cleanup(func() {
+			nq.Shutdown(ctx)
+		})
+		h := handler.NewPeriodic(func(ctx context.Context) (err error) {
+			job, err := jobs.FromContext(ctx)
+			if err != nil {
+				t.Fatalf("failed to extract job details from context: %+v", err)
+				return nil
+			}
+			_, exists := jobsProcessed.LoadOrStore(job.ID, "foo")
+			if exists {
+				t.Fatalf("job (%d) has already been processed by another worker!", job.ID)
+			}
+			atomic.AddUint32(&jobsCompleted, 1)
+			return
+		})
+
+		h.WithOptions(
+			handler.JobTimeout(500*time.Millisecond),
+			handler.Concurrency(1),
+		)
+
+		err = nq.StartCron(ctx, cron, h)
+		if err != nil {
+			t.Error(err)
+		}
+
+		workers[i] = nq
+	}
+
+	const WaitForJobTime = 1100 * time.Millisecond
+
+	// allow time for listener to start and for at least one job to process
+	time.Sleep(WaitForJobTime)
```

Comment:
Can we add a `done` channel here instead? `1.1s` is a pretty long fixed wait time. Some of my early tests did this, but I've tried to get away from any fixed wait times in favor of a `done` channel and a timeout instead.